File: my-folder/0003-longest-substring-without-repeating-characters/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l = 0
        curr_set = set()
        longest_substr = 0
        for r in range(len(s)):
            if s[r] not in curr_set:
                curr_set.add(s[r])
            else:
                while(s[r] in curr_set):
                    curr_set.remove(s[l])
                    l += 1
                curr_set.add(s[r])
            longest_substr = max(longest_substr, len(curr_set))
        
        return longest_substr
            


        

File: my-folder/0003-longest-substring-without-repeating-characters/solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
     Set<Character> wind = new HashSet<>();
     int maxWind = 0;
     int l = 0;
     for(int r = 0; r < s.length(); r++){
        char curr = s.charAt(r);
        while(wind.contains(curr)){
            wind.remove(s.charAt(l));
            l++;
        }

        wind.add(curr);
        if(r - l + 1 > maxWind){
            maxWind = Math.max(r - l + 1, maxWind);
        }
    }   

    return maxWind;
    
    }
}

// abcabcbb
//    i
// maxwind = 3
// window = {a, b, c}

